File: subprojects/common/ddo-dal/src/main/java/io/truthencode/ddo/dal/repositories/RaceFamilyRepository.java
package io.truthencode.ddo.dal.repositories;

import io.quarkus.hibernate.reactive.panache.PanacheRepository;
import io.truthencode.ddo.dal.entity.RaceFamily;
import jakarta.enterprise.context.ApplicationScoped;


/**
 * Repository for managing RaceFamily entities using Quarkus Hibernate Reactive Panache.
 * This repository provides data access layer operations for RaceFamily entities.
 */
@ApplicationScoped
public class RaceFamilyRepository implements PanacheRepository<RaceFamily> {
    /**
         * Default constructor for RaceFamilyRepository.
         * Required by JPA for creating instances of the repository.
         */
    public RaceFamilyRepository() {
        // JPA Required Default constructor
    }
}

File: subprojects/common/ddo-dal/src/main/kotlin/io/truthencode/ddo/dal/repository/RaceFamilyRepository.kt
package io.truthencode.ddo.dal.repository

import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository
import io.truthencode.ddo.dal.entity.RaceFamily
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class RaceFamilyRepository : PanacheRepository<RaceFamily> {
    fun getByName(name: String) = find("name", name).firstResult()
}